documentation/source/library_reference.rst

```diff
@@ -64,6 +64,17 @@ Interacting with the Simulator
 
 .. autofunction:: cocotb.decorators.RunningTask.kill
 
+HDL Datatypes
+-------------
+
+These are a set of datatypes that model the behavior of common HDL datatypes.
+They can be used independently of cocotb for modeling and will replace :class:`BinaryValue`
+as the types used by cocotb's `simulator handles <#simulation-object-handles>`_.
+
+.. autoclass:: cocotb.types.Logic
```

Comment:
This just shows the type docstring and nothing else since everything is a dunder method. Should we turn on showing dunder methods on these classes?

Reply:
Maybe the description of these types and their operators is easier to understand when written in a few consecutive paragraphs instead of each operator having its own docstring.

Reply:
I'll got about expanding on examples in the type docstring using `..codeblock`s.